cdap-security/src/test/java/co/cask/cdap/security/store/FileSecureStoreTest.java

```diff
@@ -77,75 +77,91 @@ public void tearDown() throws IOException {
   }
 
   @Test
   public void testOverwrite() throws IOException, InterruptedException {
-    secureStoreManager.put(KEY1, VALUE1.getBytes(Charsets.UTF_8), DESCRIPTION1, PROPERTIES_1);
-    SecureStoreData oldData = secureStore.get(KEY1);
+    secureStoreManager.put(NAMESPACE1, KEY1, VALUE1.getBytes(Charsets.UTF_8), DESCRIPTION1, PROPERTIES_1);
+    SecureStoreData oldData = secureStore.get(NAMESPACE1, KEY1);
     long oldCreateTime = oldData.getMetadata().getLastModifiedTime();
     Assert.assertArrayEquals(VALUE1.getBytes(Charsets.UTF_8), oldData.get());
     String newVal = "New value";
-    secureStoreManager.put(KEY1, newVal.getBytes(Charsets.UTF_8), DESCRIPTION1, PROPERTIES_1);
-    long newCreateTime = secureStore.get(KEY1).getMetadata().getLastModifiedTime();
-    Assert.assertArrayEquals(newVal.getBytes(Charsets.UTF_8), secureStore.get(KEY1).get());
+    secureStoreManager.put(NAMESPACE1, KEY1, newVal.getBytes(Charsets.UTF_8), DESCRIPTION1, PROPERTIES_1);
+    long newCreateTime = secureStore.get(NAMESPACE1, KEY1).getMetadata().getLastModifiedTime();
+    Assert.assertArrayEquals(newVal.getBytes(Charsets.UTF_8), secureStore.get(NAMESPACE1, KEY1).get());
     Assert.assertNotEquals(oldCreateTime, newCreateTime);
   }
 
   @Test(expected = IOException.class)
-  public void testGet_NonExistent() throws IOException {
-    secureStore.get("Dummy");
+  public void testGetNonExistent() throws IOException {
+    secureStore.get(NAMESPACE1, "Dummy");
   }
 
   @Test(expected = IOException.class)
   public void testDelete() throws IOException {
     populateStore();
     SecureStoreMetadata metadata = SecureStoreMetadata.of(KEY1, DESCRIPTION1, PROPERTIES_1);
     SecureStoreData secureStoreData = new SecureStoreData(metadata, VALUE1.getBytes(Charsets.UTF_8));
-    Assert.assertArrayEquals(secureStoreData.get(), secureStore.get(KEY1).get());
-    secureStoreManager.delete(KEY1);
+    Assert.assertArrayEquals(secureStoreData.get(), secureStore.get(NAMESPACE1, KEY1).get());
+    secureStoreManager.delete(NAMESPACE1, KEY1);
     try {
-      secureStore.get(KEY1);
+      secureStore.get(NAMESPACE1, KEY1);
     } catch (IOException ioe) {
       Assert.assertTrue(ioe.getMessage().contains("not found in the secure store"));
       throw ioe;
     }
   }
+
+  @Test
+  public void testMultipleNamespaces() throws IOException {
+    populateStore();
+    String ns = "namespace2";
+    secureStoreManager.put(ns, KEY1, VALUE1.getBytes(Charsets.UTF_8), DESCRIPTION1, PROPERTIES_1);
+    List<SecureStoreMetadata> expectedList = new ArrayList<>();
+    expectedList.add(secureStore.get(NAMESPACE1, KEY2).getMetadata());
+    expectedList.add(secureStore.get(NAMESPACE1, KEY1).getMetadata());
+    Assert.assertEquals(expectedList, secureStore.list(NAMESPACE1));
+    Assert.assertNotEquals(expectedList, secureStore.list(ns));
+    expectedList.clear();
+    expectedList.add(secureStore.get(ns, KEY1).getMetadata());
```

Comment:
However, it may be ok considering that you're using this to test the output of list, not get.